addons/web/static/src/js/chrome/abstract_web_client.js

```diff
@@ -118,6 +118,16 @@ var AbstractWebClient = Widget.extend(ServiceProviderMixin, KeyboardNavigationMi
         // If not set on the url, retrieve cids from the local storage
         // of from the default company on the user
         var current_company_id = session.user_companies.current_company[0]
+        if (config.device.isOdooMobileOS) {
+            // Native apps can be opened via native notification. In order to
+            // avoid multi-company access error, the app must be initialized
+            // with all allowed companies active. It works well on Android
+            // because it has cookie cids equal to all allowed companies.
+            // However, iOS app has session id cookie only and there is no way
+            // to update iOS app. So, as a workaround we force cids here.
+            var allowed_company_ids = session.user_companies.allowed_companies.map(value => value[0]);
+            state.cids = allowed_company_ids.join(",");
```

Comment:
Not sure what the consequence may be for the user when "a lot" of companies are now always selected...

Reply:
> Also, note that all companies are already activated on opening android app

I don't get how they may be activated. The Android app doesn't have any specific logic for that, it just keeps whatever the user has selected (using a cookie). The iOS app resets the cookies every time, hence the "only one company selected" behavior and doesn't remember what the user have chosen (which is you issue).

In regard to adding the company_id, I'm really not sure we could do it without touching the mobile apps... we'll check that later.

Reply:
On android if you close the app and start over, it will reset cookies to "all company" mode

Reply:
I don't get it... There is, AFAIK, nothing that force it to "all company" mode... and it's not the behavior in desktop either... :thinking: 

Don't know how you reproduce it ; personally I don't (at least on a clean/barebone database)

Reply:
When I open android app it shows all companies selected.
If I close the app (not just switch to home page, but remove from memory), then all companies are selected again.

Screencast: https://drive.google.com/file/d/14b497Tvf4R_jmltvXsucFFFXfmm-pYju/view?usp=sharing

Reply:
I (still) don't get it: this PR is in 13.0 and your screencast in a more recent version (15.0?).

In 13.0 (runbot-all), when checking multiple companies, after relaunching (terminate+start) the app, it rolls back to the default company only.
In 15.0, when checking multiple companies, after relaunching the app, the selected companies are restored (cookies?) but not all companies are selected (aka no "all company" mode).

So, selecting by default all companies (which can potentially add *a lot* of data in some configuration, like many companies and also confuse the user) looks risky to me, specially on a stable release.

Don't get me wrong: I see that we have limited possibilities to fix this issue without touching the iOS app...

Reply:
@pparidans you probably misunderstood:  my screencast is for android, not iOS. What I want to say, is that it should be ok to start iOS app in all company mode, because it already works this way on android (screencast is for v15 and I just checked for v13 - the same)

Reply:
I get it that your screencast is on Android... What I highlight is that I don't have the behavior you describe (aka "all company" mode) in my testing (on Android 12) ; hence the description above.

My testing was done on a clean database just by creating a few companies and allow my test user for them. Did you add some other configuration?